src/domain/accounts/services/src/use_cases/delete_account_use_case_impl.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::sync::Arc;
+
+use internal_error::InternalError;
+use kamu_accounts::*;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[dill::component]
+#[dill::interface(dyn DeleteAccountUseCase)]
+pub struct DeleteAccountUseCaseImpl {
+    current_account_subject: Arc<CurrentAccountSubject>,
+    rebac_service: Arc<dyn kamu_auth_rebac::RebacService>,
+    account_service: Arc<dyn AccountService>,
+    outbox: Arc<dyn messaging_outbox::Outbox>,
+}
+
+impl DeleteAccountUseCaseImpl {
+    async fn authenticated(&self, account_name: &odf::AccountName) -> Result<bool, InternalError> {
+        match self.current_account_subject.as_ref() {
+            CurrentAccountSubject::Anonymous(_) => Ok(false),
+            CurrentAccountSubject::Logged(l) if l.account_name == *account_name => Ok(true),
+            CurrentAccountSubject::Logged(l) => {
+                use internal_error::ResultIntoInternal;
+                use kamu_auth_rebac::RebacServiceExt;
+
+                let is_admin = self
+                    .rebac_service
+                    .is_account_admin(&l.account_id)
+                    .await
+                    .int_err()?;
+
+                Ok(is_admin)
+            }
+        }
+    }
+}
```

Comment:
Idea is great, but it does not belong to this use case. It could be a very useful shared helper.

Reply:
Extracted to `utils::AccountAuthorizationHelper`

Done